toolprovider/mise/install_plugin.go

```diff
@@ -0,0 +1,139 @@
+package mise
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/bitrise-io/bitrise/v2/log"
+	"github.com/bitrise-io/bitrise/v2/toolprovider/provider"
+)
+
+type PluginSource struct {
+	PluginName  provider.ToolID
+	GitCloneURL string
+}
+
+var pluginSourceMap = map[provider.ToolID]PluginSource{
+	"flutter": {PluginName: "flutter", GitCloneURL: "https://github.com/asdf-community/asdf-flutter.git"},
+	"tuist":   {PluginName: "tuist", GitCloneURL: "https://github.com/tuist/asdf-tuist.git"},
+}
+
+var miseCoreTools = []string{
+	"bun",
+	"deno",
+	"elixir",
+	"erlang",
+	"go",
+	"golang",
+	"java",
+	"node",
+	"nodejs",
+	"python",
+	"ruby",
+	"rust",
+	"swift",
+	"zig",
+}
+
+// InstallPlugin installs a plugin for the specified tool, if needed.
+//
+// It resolves the plugin source from the tool request or predefined map,
+// checks if the plugin is already installed, and if not, installs it using mise.
+func (m *MiseToolProvider) InstallPlugin(tool provider.ToolRequest) error {
+	if slices.Contains(miseCoreTools, string(tool.ToolName)) {
+		// Core tools do not require plugin installation.
+		return nil
+	}
+
+	plugin := parsePluginSource(tool)
+	if plugin == nil {
+		return provider.ToolInstallError{
+			ToolName:         tool.ToolName,
+			RequestedVersion: tool.UnparsedVersion,
+			Cause:            fmt.Sprintf("This tool integration (%s) is not tested or vetted by Bitrise.", tool.ToolName),
+			Recommendation:   fmt.Sprintf("If you want to use this tool anyway, look up its asdf plugin and set its git clone URL in tool_config.extra_plugins. For example: `%s: https://github/url/to/asdf/plugin/repo.git`", tool.ToolName),
+		}
+	}
+	if plugin.PluginName == "" {
+		// Plugin name is required to install the plugin.
+		return fmt.Errorf("plugin name for tool %s is not defined", tool.ToolName)
+	}
+
+	installed, err := m.isPluginInstalled(*plugin)
+	if err != nil {
+		log.Warnf("Failed to check if plugin is already installed: %v", err)
+	}
+	if installed {
+		log.Debugf("Tool plugin %s is already installed, skipping installation.", tool.ToolName)
+		return nil
+	}
+
+	pluginInstallArgs := []string{"install", string(plugin.PluginName)}
+
+	if plugin.GitCloneURL != "" {
+		pluginInstallArgs = append(pluginInstallArgs, plugin.GitCloneURL)
+	}
+
+	_, err = m.ExecEnv.RunMisePlugin(pluginInstallArgs...)
+	if err != nil {
+		return err
+	}
+
+	// Check if the plugin is found in the list of installed plugins after adding.
+	installed, err = m.isPluginInstalled(*plugin)
+	if err != nil {
+		return fmt.Errorf("check if plugin was installed successfully: %w", err)
+	}
+	if !installed {
+		return fmt.Errorf("%s plugin could not be installed", tool.ToolName)
+	}
+
+	return nil
+}
+
+func (m *MiseToolProvider) isPluginInstalled(plugin PluginSource) (bool, error) {
+	pluginListArgs := []string{"list", "--urls"}
```

Comment:
Since we parse the output manually, I think it wouldn't hurt adding the `--quiet` flag just in case: https://github.com/bitrise-io/bitrise/blob/11cbf9de408c236bbc6703d8f534483e18d51f78/toolprovider/mise/env.go#L21